src/nprintml/label/aggregator/pcap.py

```diff
@@ -48,93 +55,113 @@ def normalize_npt(cls, npt_csv, path_input_base=None):
+                header = np.genfromtxt(npt_file, delimiter=',', max_rows=1, dtype=str)
+                usecols = np.arange(1, len(header))  # ignore data index (src_ip)
+                skip_header = 1 if isinstance(npt_file, (str, pathlib.Path)) else 0
+            else:
+                skip_header = 1
+
+            npt = np.genfromtxt(
+                npt_file,
+                delimiter=',',
+                skip_header=skip_header,
+                usecols=usecols,
+                dtype=NPT_DTYPE,
+            )
+
+            npt_dim1 = len(npt.shape) == 1
+            npt_length = 1 if npt_dim1 else len(npt)
+            max_length = max(max_length, npt_length)
+
+            npt_flat = npt if npt_dim1 else npt.ravel()
+
+            yield list(itertools.chain([file_index], npt_flat))
+
+        print('Loaded', file_count, 'nprints')
+
+        return (header[1:], max_length)
+
+    @classmethod
+    def merge_npt(cls, npts_flat):
+        """Merge nPrint data from multiple output files."""
+        npts = pd.DataFrame(npts_flat, dtype=NPT_DTYPE)
+
+        # true index is generated from input; rather than iterate
+        # multiple times, it's initially treated as data.
+        # now we can move it in place:
+        npts.set_index(0, inplace=True)
+
+        # npt lengths are unknown ahead of time; and, with flatten(), each becomes a row length.
+        # luckily, we need not make them equal first -- pandas fills in na or None.
+        # constructor doesn't allow specification of this missing sentinel;
+        # here we set it:
+        npts.fillna(-1, inplace=True)
+
+        # though column names might not be TOO important they may be helpful.
+        # here we flatten these as well from the maximum size we might require:
+        (header, max_length) = npts_flat.result
+        if header is not None:
+            npts.columns = cls.flatten_columns(header, max_length)
+
+        print('nPrints padded to maximum size:', max_length)
+        print('nPrint features shape:', npts.shape)
+
+        return npts
+
     def __init__(self, label_csv):
         super().__init__(label_csv)
         self.labels = self.load_label(label_csv)
 
-    def __call__(self, npt_csv, path_input_base=None, compress=False, sample_size=1):
+    def __call__(self, npt_csv, compress=False, sample_size=1, path_input_base=None):
         """Enumerate given stream or directory of nPrint data, load,
         pad to their maximum size, and attach labels.
 
         `npt_csv` may specify either a path to a directory of nPrint
         results (`str` or `pathlib.Path`) OR a stream of open file-like
         objects (exposing both `read()` and `name`).
 
-        `path_input_base` is recommended when `npt_csv` specifies a
-        stream of file objects, to indicate their common base path (even
-        if this is virtual), such that they may be matched with the
-        label index.
+        `path_input_base` is suggested when `npt_csv` specifies a stream
```

Comment:
(You'll see below that I simplified some previous work such that some of this `path_input_base` stuff is no longer required.)